File: check_firestore.js
const admin = require('firebase-admin');
const serviceAccount = require('./backend/service-account-key.json');

// Firebase Admin 초기화
admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  storageBucket: 'credible-runner-474101-f6.firebasestorage.app'
});

const db = admin.firestore();

async function checkFirestore() {
  const userId = 'ZH4dY6r3y3fbABpoCVtVbDqqqzG3';
  const callId = 'test_call_1760593050419';

  try {
    console.log('📊 Checking Firestore for call document...');
    console.log('   Path: users/' + userId + '/calls/' + callId);

    const docRef = db.collection('users').doc(userId).collection('calls').doc(callId);
    const doc = await docRef.get();

    if (!doc.exists) {
      console.log('❌ Document not found!');
      return;
    }

    console.log('\n✅ Document found!');
    console.log('\n📄 Document data:');
    const data = doc.data();

    // Pretty print
    console.log(JSON.stringify(data, null, 2));

    console.log('\n🔍 Key fields:');
    console.log('   status:', data.status);
    console.log('   analysisStatus:', data.analysisStatus);
    console.log('   fileName:', data.fileName);
    console.log('   filePath:', data.filePath);
    console.log('   updatedAt:', data.updatedAt?.toDate?.());

  } catch (error) {
    console.error('❌ Error checking Firestore:', error);
    throw error;
  }
}

checkFirestore()
  .then(() => process.exit(0))
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });

File: scripts/check_firestore.py
#!/usr/bin/env python3
"""
Check Firestore Call Document
"""

from google.cloud import firestore

PROJECT_ID = 'my-project-54928-b9704'
USER_ID = '7wll6D15YZgVrL7jEO1dJhyCUKG3'
CALL_ID = 'test_call_1760574708863'

def check_call_document():
    db = firestore.Client(project=PROJECT_ID)

    doc_ref = db.collection('users').document(USER_ID).collection('calls').document(CALL_ID)
    doc = doc_ref.get()

    if doc.exists:
        print("=" * 60)
        print(f"📄 Call Document: {CALL_ID}")
        print("=" * 60)
        data = doc.to_dict()
        for key, value in sorted(data.items()):
            print(f"{key:20s}: {value}")
        print("=" * 60)
    else:
        print(f"❌ Document not found: {CALL_ID}")

if __name__ == "__main__":
    check_call_document()
